crates/holochain_cascade/src/lib.rs

```diff
@@ -993,19 +993,50 @@ where
         })
             .await??;
 
-        // For each response run the chain filter and check the invariants hold.
-        for response in results {
-            let result =
-                authority::get_agent_activity_query::must_get_agent_activity::filter_then_check(
-                    response,
-                )?;
-
-            // Short circuit if we have a result.
-            if matches!(result, MustGetAgentActivityResponse::Activity(_)) {
-                return Ok(result);
+        let mut merged_results = (MustGetAgentActivityResponse::EmptyRange, None);
+
+        for result in &results {
```

Comment:
@jost-s i think it looks more complicated than it is because of the pattern matching, it either does nothing, keeps what it already has, or merges new results if they are compatible

happy to add comments tho